### PR TITLE
CT-647: ETH Batch Transactions

### PR DESCRIPTION
**Pull request by bpagon13**
This feature branch grants the SDK the ability to validate ETH batch transactions. The majority of the code has already been reviewed in https://github.com/BitGo/BitGoJS/pull/444, with a few changes made to fix merge conflicts between that PR and master.

### CODE REVIEW

**Comment by bpagon13** [comment]
Failing test `should create an XRP address` is unrelated to changes in this PR, caused by reset of XRP testnet.

File: modules/core/src/v2/coins/eth.ts [hunk 1]
```diff
@@ -1196,6 +1208,119 @@ export class Eth extends BaseCoin {
...
+  }
+
+  /**
+   * Validates that the prebuild data returned by platform is valid
+   * @param params
+   * @param callback
+   * @throws Error if the prebuild is invalid
+   */
+  verifyTransaction(params: any, callback?: NodeCallback<any>): Bluebird<any> {
+    return co(function*() {
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/eth.ts]: add `const self = this;` above coroutine.

File: modules/core/src/v2/coins/eth.ts [hunk 2]
```diff
@@ -1196,6 +1208,119 @@ export class Eth extends BaseCoin {
...
+          `prebuild recipients array should always be of length 1, was of length ${txPrebuild.recipients.length} here`
+        );
+      }
+      if (txParams.recipients.length === 1 && !_.isEqual(txPrebuild.recipients, txParams.recipients)) {
+        throw new InvalidEthereumRecipients(
+          'recipients returned by BitGo are not equal to recipients in original tx request'
+        );
+      }
+      if (txParams.recipients.length > 1) {
+        this.validateBatchData(txPrebuild.recipients, txParams.recipients);
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/eth.ts]: Typescript has trouble understanding the type of `this` inside of coroutines (long story). To help it out, we usually declare a new variable `self` and assign it to `this` at the  start of the function, then  use `self` throughout.  This helps the typescript compiler  check types on calls involving `this`, which it otherwise couldn't do.

TL;DR: this should be `self.validateBatchData()`.

File: modules/core/src/v2/environments.ts [hunk 3]
```diff
@@ -63,6 +64,7 @@ const mainnetBase: EnvironmentTemplate = {
     'xpub661MyMwAqRbcEtUgu9HF8ai4ipuVKKHBzUqks4jSFypW8dwwQL1zygLgQx99NmC7zJJznSiwKG6RQfVjAKMtCsx8VjR6kQW8x7HrkXFZdnQ',
   hsmXpub:
     'xpub661MyMwAqRbcGEtTFgMAoxMFoqsRdDaiaR63byNfZdV7cBZFvovQSNJ5bpyeoQtuKVgXBk6sFQ7TtvyWqadt41DnKwveYgM5KoU2EKYjdY2',
+  batcherContractAddress: undefined, // TODO [CT-928] replace with mainnet Batcher contract address
```
Comment (tylerlevine) [inline modules/core/src/v2/environments.ts]: Do we know what this is yet?

**Comment by tylerlevine** [comment]
Also the failing xrp test should be disabled now, so please rebase to pick that up.

File: modules/core/src/v2/environments.ts (hunk 3 above)
Comment (bpagon13) [reply modules/core/src/v2/environments.ts]: No, we don't yet. I can open up another PR to fill this in when the Batcher is deployed. Right now, the platform will block ETH batch transactions until it's deployed.

File: modules/core/src/v2/coins/eth.ts (hunk 1 above)
Comment (bpagon13) [reply modules/core/src/v2/coins/eth.ts]: Fixed.

File: modules/core/src/v2/coins/eth.ts (hunk 2 above)
Comment (bpagon13) [reply modules/core/src/v2/coins/eth.ts]: Fixed.
